File: src/main/java/com/aquasecurity/plugins/trivy/ui/TrivyWindowFactory.java
package com.aquasecurity.plugins.trivy.ui;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;

public class TrivyWindowFactory implements ToolWindowFactory {

    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        TrivyWindow TrivyWindow = new TrivyWindow(project);
        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
        Content content = contentFactory.createContent(TrivyWindow.getContent(), "", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/ui/TrivyWindowFactory.kt
package com.aquasecurity.plugins.trivy.ui

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory
import com.intellij.ui.content.ContentFactory

class TrivyWindowFactory : ToolWindowFactory {
  override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
    val TrivyWindow = TrivyWindow(project)
    val contentFactory = ApplicationManager.getApplication().getService(ContentFactory::class.java)
    val content = contentFactory.createContent(TrivyWindow.content, "", false)
    toolWindow.contentManager.addContent(content)
  }
}
